### PR TITLE
[doc] Add doc about difference between taichi and python programs

### PR DESCRIPTION
**Pull request by lin-hitonami**
Related issue = #

### CODE REVIEW

File: docs/lang/articles/basic/difference_between_taichi_and_python_programs.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+sidebar_position: 2
+---
+
+# Difference between taichi and python programs
+
+Although taichi uses Python as the frontend, there are some differences between taichi and python programs.
```
Comment (k-ye) [inline docs/lang/articles/basic/difference_between_taichi_and_python_programs.md:7]: nit: Taichi

File: docs/lang/articles/basic/difference_between_taichi_and_python_programs.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+1. Taichi only supports one return statement outside non-static `if`/`for`/`while` scope in the program while Python supports return statements in other places.
+2. Taichi uses lexical scoping (static scoping) while python uses dynamic scoping.
+3. Taichi does not support some of Python's language features.
+
+## Return statement
+
+- If a taichi kernel/function does not have a return statement, it must not have return type annotation.
+- If a taichi kernel has a return statement, it must have return type annotation.
+- If a taichi function has a return statement, return type annotation is optional.
```
Comment (k-ye) [inline docs/lang/articles/basic/difference_between_taichi_and_python_programs.md:18]: Let's not suggest this. I believe type annotation for functions will be equally important once we support real function.

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

File: docs/lang/articles/basic/differences_between_taichi_and_python_programs.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
+---
+sidebar_position: 2
+---
+
+# Differences between Taichi and Python programs
+
+Although Taichi uses Python as the frontend, there are some differences between Taichi and Python programs.
+Main differences are:
```
Comment (Leonz5288) [inline docs/lang/articles/basic/differences_between_taichi_and_python_programs.md:8]: Might sounds better with "Major differences:"

File: docs/lang/articles/basic/differences_between_taichi_and_python_programs.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+2. [Variables defined inside an `if`/`for`/`while` block cannot be accessed outside the block.](#variable-scoping)
+3. [Taichi does not fully support some language features of Python.](#unsupportedpartially-supported-python-language-features)
+  - [Set, list, dictionary and operator `in`](#set-list-dictionary-and-operator-in)
+  - [Comprehensions](#comprehensions)
+  - [Operator `is`](#operator-is)
+
+## Return statement and return type annotation
+
+- If a taichi kernel/function does not have a return statement, it must not have return type annotation.
+- If a taichi kernel has a return statement, it must have return type annotation.
```
Comment (Leonz5288) [inline docs/lang/articles/basic/differences_between_taichi_and_python_programs.md:20]: These two lines are somewhat repeating? May be better with:

In a Taichi kernel, return statement must appear with the existence of return type annotation, vice versa.

File: docs/lang/articles/basic/differences_between_taichi_and_python_programs.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+1. [Taichi only supports return statement outside non-static `if`/`for`/`while` scope in the program](#return-statement)
+2. [Variables defined inside an `if`/`for`/`while` block cannot be accessed outside the block.](#variable-scoping)
+3. [Taichi does not fully support some language features of Python.](#unsupportedpartially-supported-python-language-features)
+  - [Set, list, dictionary and operator `in`](#set-list-dictionary-and-operator-in)
+  - [Comprehensions](#comprehensions)
+  - [Operator `is`](#operator-is)
+
+## Return statement and return type annotation
+
+- If a taichi kernel/function does not have a return statement, it must not have return type annotation.
```
Comment (Leonz5288) [inline docs/lang/articles/basic/differences_between_taichi_and_python_programs.md:19]: I believe 'taichi' has to be capitalized as 'Taichi'?

File: docs/lang/articles/basic/differences_between_taichi_and_python_programs.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+## Unsupported/partially supported Python language features
+
+### Set, list, dictionary and operator `in`
+
+Currently, Taichi does not support `set`.
+
+List and dictionary before assigning to a variable works as the python list and dictionary.
+However, after assigning to a variable, the content of the list and the values (not keys) of the dictionary are converted to Taichi variables.
+
+Taichi does not have a runtime implementation of `in` currently. So, operator `in` and `not in` only works in  [static scope](/lang/articles/advanced/meta#static-scope) (inside `ti.static()`).
```
Comment (Leonz5288) [inline docs/lang/articles/basic/differences_between_taichi_and_python_programs.md:127]: `So` -> `Therefore`
`So` cannot connect two complete sentences.

File: docs/lang/articles/basic/differences_between_taichi_and_python_programs.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+@ti.kernel
+def error_non_static_in():
+    if i in [1, 2]:  # Error: Cannot use `in` outside static scope
+        pass
+```
+
+### Comprehensions
+
+Taichi partially supports list comprehension and dictionary comprehension,
+and does not support set comprehension.
```
Comment (Leonz5288) [inline docs/lang/articles/basic/differences_between_taichi_and_python_programs.md:156]: `but` might be better than `and`?

**Review by Leonz5288 — COMMENTED** [review]
Some suggestions on texts.

File: docs/lang/articles/basic/differences_between_taichi_and_python_programs.md (hunk 4 above)
Comment (Leonz5288) [reply docs/lang/articles/basic/differences_between_taichi_and_python_programs.md]: Make sense

**Review by Vissidarte-Herman — CHANGES_REQUESTED** [review]

**Review by ailzhang — APPROVED** [review]
LGTM!

**Pull request merged**